client/services.go

```diff
@@ -0,0 +1,154 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/systemd"
+)
+
+// ServicesOp encapsulate requests for performing an operation on a series of services
+type ServiceOp struct {
+	Services []string `json:"services,omitempty"`
+	Action   string   `json:"action"`
+}
+
+func (op ServiceOp) Description() string {
+	var verb string
+	switch op.Action {
+	case "enable-now":
+		verb = i18n.G("Enable and start")
+	case "disable-now":
+		verb = i18n.G("Stop and disable")
+	case "try-reload-or-restart":
+		verb = i18n.G("Try to reload or restart")
+	case "reload":
+		// the following are spelled out so xgettext finds them
+		verb = i18n.G("Reload")
+	case "start":
+		verb = i18n.G("Start")
+	case "stop":
+		verb = i18n.G("Stop")
+	case "enable":
+		verb = i18n.G("Enable")
+	case "disable":
+		verb = i18n.G("Disable")
```

Comment:
These two should also be part of start/stop, per agreements. Separating them one means a request such as "snap stop --disable foo" might half-work.